system/lib/libc/musl/src/stdio/__stdio_seek.c

```diff
@@ -4,7 +4,13 @@ off_t __stdio_seek(FILE *f, off_t off, int whence)
 {
 	off_t ret;
 #ifdef SYS__llseek
-	if (syscall(SYS__llseek, f->fd, off>>32, off, &ret, whence)<0)
+	off_t off_hi =
+#ifndef __EMSCRIPTEN__
+		off >> 32;
```

Comment:
Indeed.

>If the value of the right operand is negative or is greater than or equal to the width of the promoted left operand, the behavior is undefined.

The JS backend lowers this to asm as explicitly storing 0 in that slot in the vararg buffer that gets passed in.
The wasm backend lowers this by not storing anything to that slot in the vararg buffer, meaning it gets whatever was on the stack at that address beforehand, aka probably-not-zero.

Amusingly, with non-vararg syscalls (https://github.com/kripken/emscripten/pull/5151) the wasm backend passes the stack pointer as its `undef` value, where js backend still passes 0. I'm guessing because LRU virtual register? Not that why matters too much because `undef`.